packages/xstate-wallet/src/integration-tests/basic.test.ts

```diff
@@ -0,0 +1,170 @@
+  isNotification,
+  PushMessageRequest,
+  CreateChannelResponse,
+  JoinChannelRequest,
+  JoinChannelResponse
+} from '@statechannels/client-api-schema';
+import {filter, map} from 'rxjs/operators';
+import {Observable} from 'rxjs';
+import waitForExpect from 'wait-for-expect';
+
+jest.setTimeout(30000);
+
+function generatePushMessage(data: any, recipient: string, sender: string): PushMessageRequest {
+  return {
+    jsonrpc: '2.0',
+    id: 111111111,
+    method: 'PushMessage',
+    params: {data, recipient, sender}
+  };
+}
+
+function generateJoinChannelRequest(channelId: string): JoinChannelRequest {
+  return {id: 222222222, method: 'JoinChannel', jsonrpc: '2.0', params: {channelId}};
+}
+
+function generateCreateChannelRequest(
+  playerA: Participant,
+  playerB: Participant
+): CreateChannelRequest {
+  return {
+    jsonrpc: '2.0',
+    id: 3333333333,
+    method: 'CreateChannel',
+    params: {
+      participants: [playerA, playerB],
+      allocations: [
+        {
+          token: '0x0',
+          allocationItems: [
+            {
+              destination: playerA.destination,
+              amount: '0x06f05b59d3b20000'
+            },
+            {
+              destination: playerA.destination,
+              amount: '0x06f05b59d3b20000'
+            }
+          ]
+        }
+      ],
+      appDefinition: '0x430869383d611bBB1ce7Ca207024E7901bC26b40',
+      appData: '0x0' // TODO: This works for now but will break when we start validating
+    }
+  };
+}
+
+it('works', async () => {
+  const playerA = new Player('0x275a2e2cd9314f53b42246694034a80119963097e3adf495fbf6d821dc8b6c8e');
+  const playerB = new Player('0x3341c348ea8ade1ba7c3b6f071bfe9635c544b7fb5501797eaa2f673169a7d0d');
+
+  playerA.channelWallet.onSendMessage(message => {
+    if (isNotification(message) && message.method === 'MessageQueued') {
+      const pushMessageRequest = generatePushMessage(
+        message.params,
+        playerB.participantId,
+        playerA.participantId
+      );
+      playerB.channelWallet.pushMessage(pushMessageRequest);
+    }
+  });
+  playerB.channelWallet.onSendMessage(message => {
+    if (isNotification(message) && message.method === 'MessageQueued') {
+      const pushMessageRequest = generatePushMessage(
+        message.params,
+        playerA.participantId,
+        playerB.participantId
+      );
+      playerA.channelWallet.pushMessage(pushMessageRequest);
+    }
+  });
+
+  const createEvent = generateCreateChannelRequest(playerA.participant, playerB.participant);
+
+  const playerAResponsePromise = createPromise(
+    playerA.messagingService.outboxFeed.pipe(
+      filter(m => 'id' in m && m.id === createEvent.id),
+      map(m => m as CreateChannelResponse)
+    )
+  );
+
+  await playerA.messagingService.receiveMessage(createEvent);
+
+  playerA.channelWallet.workflows[0].machine.send({type: 'USER_APPROVES'});
+
+  const response = await playerAResponsePromise;
+  expect(response.result).toBeDefined();
```

Comment:
Can we be specific about the response result?

Reply:
Yup, updated.